Sources/GoodExtensions/UIKit/UIViewExtensions.swift

```diff
@@ -254,36 +255,53 @@ public extension GRActive where Base: UIView {
 
 }
 
-// MARK: - Blur
+
+// MARK: - Blurring
 
 public extension GRActive where Base: UIView {
+    
+    private final class BlurredImageView: UIImageView, NameDescribable {}
 
-    func blur(_ blurRadius: Double = 3.5) {
+    func blur(_ blurRadius: Double = 3.5, duration: CGFloat = 0.3) {
         unblur()
         guard let blurredImage = createBlurryImage(blurRadius) else {
             return
         }
 
-        let blurredImageView = UIImageView(image: blurredImage)
+        let blurredImageView = BlurredImageView(image: blurredImage)
         blurredImageView.translatesAutoresizingMaskIntoConstraints = false
-        blurredImageView.tag = -419
         blurredImageView.contentMode = .center
         blurredImageView.backgroundColor = .clear
 
+        blurredImageView.alpha = 0
+        
         base.addSubview(blurredImageView)
+        
         NSLayoutConstraint.activate([
             blurredImageView.centerXAnchor.constraint(equalTo: base.centerXAnchor),
             blurredImageView.centerYAnchor.constraint(equalTo: base.centerYAnchor)
         ])
+        
+        UIView.animate(withDuration: duration) {
+            blurredImageView.alpha = 1
+        }
     }
 
-    func unblur() {
-        base.subviews.forEach {
-            if $0.tag == -419 {
-                $0.removeFromSuperview()
-                base.layoutSubviews()
-            }
+    func unblur(duration: CGFloat = 0.3) {
```

Comment:
duation by som dal default 0, defaultne by to tak bolo bez animacie. Rovnako plati pre blur.

Reply:
bud tak alebo tam dat este bol parameter animated co ty nato?